src/main/java/com/spotify/folsom/client/Request.java

```diff
@@ -65,4 +72,32 @@ protected static ByteBuf toBuffer(final ByteBufAllocator alloc, ByteBuffer dst,
   public int getOpaque() {
     return opaque;
   }
+
+  @VisibleForTesting
+  static byte[] encodeKey(String key, Charset charset) {
+    checkNotNull(key, "key");
+    byte[] keyBytes = key.getBytes(charset);
+    int length = keyBytes.length;
+    if (length > 250) {
+      throw new IllegalArgumentException("Key is too long: " + key);
+    }
+    if (length <= 0) {
+      throw new IllegalArgumentException("Key is empty");
+    }
+    for (int i = 0; i < length; i++) {
+      final byte c = keyBytes[i];
+      if (c >= 0 && c <= 32) {
+        throw new IllegalArgumentException("Invalid key: " + key);
```

Comment:
Or even more common I guess - any character represented as 0x00?? (is there where the regular ASCII fits in UTF-16?)